src/main/java/mat/client/shared/CQLLibraryResultTable.java

```diff
@@ -107,27 +76,27 @@ public void onBrowserEvent(Cell.Context context, Element elem, CQLLibraryDataSet
             public SafeHtml getValue(CQLLibraryDataSetObject object) {
                 return CellTableUtility.getColumnToolTip(MeasureDetailsUtil.defaultTypeIfBlank(object.getLibraryModelType()));
             }
-
-            /*
-                Single Click to select row and enable checkbox
-                Double Click to navigate to CQL Composer tab
-            */
-            @Override
-            public void onBrowserEvent(Cell.Context context, Element elem, CQLLibraryDataSetObject object, NativeEvent event) {
-                if (object.getLastClick() < System.currentTimeMillis() - DELAY_TIME) {
-                    object.setSelected(!object.isSelected());
-                    selectionModel.setSelected(object, object.isSelected());
-                } else if (object.isFhirEditOrViewable()) {
-                    SelectionEvent.fire(fireEvent, object);
-                }
-                object.setLastClick(System.currentTimeMillis());
-            }
         };
         if (MatContext.get().getFeatureFlagStatus(FeatureFlagConstant.MAT_ON_FHIR))
             table.addColumn(model, SafeHtmlUtils.fromSafeConstant("<span title='Version'>" + "Model" + "</span>"));
 
+        table.addCellPreviewHandler(event -> {
+            String eventType = event.getNativeEvent().getType();
+            obj = event.getValue();
+            event.getNativeEvent().preventDefault();
+            if (BrowserEvents.CLICK.equalsIgnoreCase(eventType)) {
+                if(previousObj == null){
+                    clickEventsOnSelectedRows(event, selectionModel, obj, fireEvent);
+                } else if (obj.getId() == previousObj.getId()) {
```

Comment:
Is it the same logic as for the Measures Gird?

Reply:
No, Here MultiSelectionModel isn't working as expected. It doesn't set the value as expected. Therefore, I have to handle the selected objects manually.

Reply:
for example, When we have MultiSelectionModel.setSelected as false. It immediately overrides it to true. and I couldn't see any other setSelected method being called other than what I intended to. Also selectionModel is a local variable, So I am not sure what is causing it to alter the value.